### PR TITLE
Pull API Testers into their own projects

### PR DESCRIPTION
**Pull request by beylmk**
Pulls APITesters into separate projects, due to some SPM issues. Makes both of them dependent on a branch rather than local Purchases.

Also includes changes from my [previous PR](https://github.com/RevenueCat/purchases-ios/pull/822):
* creates a new API tester for Swift
* adds error code testing to objc api tester
* updates enum testing to switches for objc api tester

The idea is to use these API testers to test our script, but eventually we can have them build off of the current branch for PRs to alert to any breaking changes.

### CODE REVIEW

**Comment by aboedo** [comment]
could you elaborate on the motivation? 
Asking because I've been considering doing the exact opposite with our integration tests, since they're awkward to test because they're in a different project. 
Also, are we considering having a workspace that integrates them all?

File: .circleci/config.yml [hunk 1]
```diff
@@ -148,24 +148,24 @@ jobs:
...
-          name: Build APITesters
-          command: xcodebuild -target SwiftAPITester
+          name: Build SwiftAPITester
+          command: xcodebuild -workspace APITesters/APITesters.xcworkspace -scheme SwiftAPITester
 
   objc_apitest:
     macos:
       xcode: "12.5.1"
-    working_directory: ~/purchases-ios/APITesters.ObjCAPITester
+    working_directory: ~/purchases-ios
```
Comment (beylmk) [inline .circleci/config.yml:162]: just curious, why is this different from changing the working directory and then running the same command `xcodebuild -workspace APITesters.xcworkspace -scheme SwiftAPITester`?

**Comment by beylmk** [comment]
> could you elaborate on the motivation?
> Asking because I've been considering doing the exact opposite with our integration tests, since they're awkward to test because they're in a different project.

Part of the motivation was to more closely mimic an end-user's setup, and to avoid a potential SPM bug that we THINK might have to do with importing a framework from within the framework's project...I'll ping you in the Slack thread about it in case you have thoughts.

> Also, are we considering having a workspace that integrates them all?

I have the APITesters workspace, do you mean consolidating the integration tests with that? Definitely open to it... I have run into weird errors with conflicting dependences in the same workspace, though.

File: .circleci/config.yml (hunk 1 above)
Comment (taquitos) [reply .circleci/config.yml:162]: I think it's exactly the same, but also your working directory was `APITesters.ObjCAPITester` and should have been `APITesters` 😄
I prefer to keep my commands from the root of the project, purely just a preference.

**Review by taquitos — APPROVED** [review]
🐐

**Comment by taquitos** [comment]
Merging for now, we can continue to have conversations around https://github.com/RevenueCat/purchases-ios/pull/850#issuecomment-918398225.

**Pull request merged**